bpf/lib/lb.h

```diff
@@ -477,6 +477,65 @@ static __always_inline int lb6_xlate(struct __ctx_buff *ctx,
 	return CTX_ACT_OK;
 }
 
+#ifdef ENABLE_SESSION_AFFINITY
+static __always_inline
+__u32 lb6_affinity_backend_id(__u16 rev_nat_id,
+			      __u32 svc_affinity_timeout,
+			      bool netns_cookie,
+			      union v6addr *client_id)
+{
+	__u32 now = bpf_ktime_get_sec();
+	struct lb_affinity_match match = { .rev_nat_id = rev_nat_id };
+	struct lb_affinity_val *val;
+	struct lb6_affinity_key key = {	.netns_cookie = netns_cookie,
+					.rev_nat_id = rev_nat_id };
+	ipv6_addr_copy(&key.client_id, client_id);
+
+
+	val = map_lookup_elem(&LB6_AFFINITY_MAP, &key);
+	if (val != NULL) {
+		if ((val->last_used + svc_affinity_timeout) < now) {
+			map_delete_elem(&LB6_AFFINITY_MAP, &key);
+			return 0;
+		}
+
+		match.backend_id = val->backend_id;
+		if (map_lookup_elem(&LB_AFFINITY_MATCH_MAP, &match) == NULL) {
+			map_delete_elem(&LB6_AFFINITY_MAP, &key);
```

Comment:
Do we really need this extra bookkeeping? It feels a bit convoluted and overly complex. Why can't we simply just store the result of the first `key.slave = (sock_local_cookie(ctx) % svc->count) + 1` in the affinity map and then reuse this in `slave_svc = __lb6_lookup_slave(&key)` instead? This way we always have a consistent view on the svc map and this extra map can be avoided entirely?

Reply:
Unfortunately, we can't store a `slave` number in the affinity map, as the position might change.